### PR TITLE
FAI-381: Score Cards: Changes to unmarshalled Characteristics not producing immer patches

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-381

Test criteria in JIRA.

### CODE REVIEW

**Review by sjamboro — APPROVED** [review]

**Comment by manstis** [comment]
@tiagobento This PR remains steadfastly keen to remain red.

I hope by the mere addition of this comment the next build will be green.. "Sods Law" has a tendency to work this way.

**Comment by manstis** [comment]
Hi @kelvah, @tiagobento could you please cast your eyes over this when you have time.

**Review by kelvah — APPROVED** [review]

File: packages/pmml-editor-marshaller/src/marshaller/Marshaller.ts [hunk 1]
```diff
@@ -93,9 +97,23 @@ function merge(arg: any[]): any {
...
+function clone(obj: any) {
+  if (obj === null || typeof obj !== "object") {
+    return obj;
+  }
+  const temp: any = new obj.constructor(obj);
+  /* tslint:disable:forin */
+  for (const key in obj) {
+    temp[key] = clone(obj[key]);
+  }
+  /* tslint:enable:forin */
```
Comment (tiagobento) [inline packages/pmml-editor-marshaller/src/marshaller/Marshaller.ts]: Should we disable that on our TSlint configuration directly?

Comment (manstis) [reply packages/pmml-editor-marshaller/src/marshaller/Marshaller.ts]: That, my friend, is up to you. `pmml-editor-marshaller` simply follows the TSLint rules for this repository.

Comment (tiagobento) [reply packages/pmml-editor-marshaller/src/marshaller/Marshaller.ts]: Well, TSLint is there to HELP, not to get in the way, so if it's getting in the way, I believe we should change its constraints. Could you please go ahead and disable this check on the tslint configuration file? =)

**Comment by manstis** [comment]
@tiagobento `tslint.json` updated as requested and _local_ disabling removed in new commit.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**